pontoon/base/static/js/translate.js

```diff
@@ -256,6 +256,21 @@ var Pontoon = (function (my) {
         success: function(data) {
           if (data.length) {
             $.each(data, function() {
+              var approveButtonTitle = "";
+              /* Moved the logic from ternary operators to make this a little bit easier to read. */
+              if (this.approved) {
+                if (this.approved_user) {
+                  approveButtonTitle = 'Approved by ' + this.approved_user;
+                } else {
+                  approveButtonTitle = 'Unapprove';
+                }
+              } else {
+                if (this.unapproved_user) {
+                  approveButtonTitle = 'Unapproved by ' + this.unapproved_user;
+                } else {
+                  approveButtonTitle = 'Approve';
+                }
+              }
```

Comment:
This is a nice simplification! I'd simplify it even further by always showing `Approve`/`Unapprove` in the button title.

`Approved/Unapproved by` can instead by moved to the `title` tag of the translation author `<a>` element (line 285).

Also, please make sure that `<button>` and `<a> `title` tags are updated after (un)approval.